documentation/book/assembly-kafka-connect-external-configuration.adoc

```diff
@@ -0,0 +1,24 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-deployment-configuration-kafka-connect.adoc
+// assembly-deployment-configuration-kafka-connect-s2i.adoc
+
+// Save the context of the assembly that is including this one.
+// This is necessary for including assemblies in assemblies.
+// See also the complementary step on the last line of this file.
+
+[id='assembly-kafka-connect-external-configuration-{context}']
+
+= Using external configuration and secrets
+
```

Comment:
@laidan6000 I fixed all the other comments as you suggested. But I'm a bit unsure about this one as I don't think it is correct and makes really sense. 

* The REST and HTTP are the same interface. Technically it should be probably _HTTP REST interface_
* The way this works is that the configuration is stored inside Kafka. And the user creates / modifies / deletes the connector and its configuration using the HTTP REST interface.
* After this PR the connector configuration is not really stored in the config maps or secrets. But the configuration in the HTTP REST commands can reference the values from them and that way to keep them separate and more secure if needed.

I think especially the first paragraph in your suggestion is wrong and confusing. The second could be probably used.

Reply:
Thanks for the additional information in the bullet points. In my eagerness to use the verb "store" rather than "externalize", I changed the meaning of the sentence. To be honest, I am struggling to fully understand the approach here. I've edited the first paragraph based on your bullet points -- does it work for you? Please tweak the wording as needed.

==

Kafka Connect connectors are configured using an HTTP REST interface. The connector configuration is stored within Kafka itself and passed to Kafka Connect as part of an HTTP request.

Alternatively, the configuration of a Kafka Connect connector can be externalized as ConfigMaps or Secrets. You can then reference the configuration values in HTTP REST commands (this keeps the configuration separate and more secure, if needed). This method applies especially to confidential data, such as usernames, passwords, or certificates.

ConfigMaps and Secrets are standard {ProductPlatformName} resources used for Pod configuration and authentication.

Reply:
It reads very well and makes sense to me. Happy for you to merge the PR.